addons/point_of_sale/models/pos_payment.py

```diff
@@ -18,9 +18,10 @@ class PosPayment(models.Model):
     _inherit = ['pos.load.mixin']
 
     name = fields.Char(string='Label', readonly=True)
-    pos_order_id = fields.Many2one('pos.order', string='Order', required=True, index=True)
-    amount = fields.Monetary(string='Amount', required=True, currency_field='currency_id', help="Total amount of the payment.")
-    payment_method_id = fields.Many2one('pos.payment.method', string='Payment Method', required=True)
+    pos_order_id = fields.Many2one('pos.order', string='Order')
+    amount = fields.Monetary(string='Amount', required=True, currency_field='currency_id', help="Total amount of the payment.", default=0)
+    # payment_method_id = fields.Many2one('pos.payment.method', string='Payment Method', required=True)
+    payment_method_id = fields.Many2one('pos.payment.method', string='Payment Method')
```

Comment:
if we go immediately for the idea of atomic call, we won't need to make changes in the schema.